### PR TITLE
Update after digest

### PR DESCRIPTION
**Pull request by carver**
## What was wrong?

Pycryptodome was not allowing update after digest. This made it difficult to use in py-evm.

## How was it fixed?

Turned on the flag in init: `update_after_digest`

Also, an unrelated fix that seems to have been broken in master?

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: tests/backends/test_results.py [hunk 1]
```diff
@@ -43,25 +43,29 @@ def test_keccak_256(keccak, prehash, expected_result):
...
+        pytest.param(
             [bytearray(b'arst'), b'tsra'],
             b"\xb1\xf3T\xb2\x8f\xf2\x84R\xd6\xb9\xd6\x1fA\x06\x1b\xbe\x82\xbe\xb1\xfc\x98\xf33d\xa8\x05\x8d\x1a]\x16M\x05",  # noqa: E501
+            marks=pytest.mark.xfail(reason="pycryptodome does not support updating w/ bytearrays"),
         ),
-        (
+        pytest.param(
             [b'arst', bytearray(b'tsra')],
             b"\xb1\xf3T\xb2\x8f\xf2\x84R\xd6\xb9\xd6\x1fA\x06\x1b\xbe\x82\xbe\xb1\xfc\x98\xf33d\xa8\x05\x8d\x1a]\x16M\x05",  # noqa: E501
+            marks=pytest.mark.xfail(reason="pycryptodome does not support updating w/ bytearrays"),
```
Comment (pipermerriam) [inline tests/backends/test_results.py]: Can't we normalize at the `eth-hash` level?

Comment (carver) [reply tests/backends/test_results.py]: you mean stop accepting bytearrays for any backend?

Comment (pipermerriam) [reply tests/backends/test_results.py]: no, I mean wrap the input value in `bytes` to coerce `bytearray -> bytes` before it gets to pycryptodome.

Comment (carver) [reply tests/backends/test_results.py]: Actually, it's moot, because my problem was a local old install of pycryptodome. False alarm.

**Pull request merged**